tap_zuora/client.py

```diff
@@ -14,109 +18,180 @@
 
     @staticmethod
-    def from_config(config):
-        sandbox = config.get('sandbox', False) == 'true'
-        european = config.get('european', False) == 'true'
-        partner_id = config.get('partner_id', None)
-        return Client(config['username'], config['password'], partner_id, sandbox, european)
-
-    def get_url(self, rest):
+    def from_config(config: Dict):
+        sandbox = config.get("sandbox", False) == "true"
+        european = config.get("european", False) == "true"
+        partner_id = config.get("partner_id", None)
+        is_rest = config.get("api_type") == "REST"
+        return Client(
+            config["username"], config["password"], partner_id, sandbox, european, is_rest
+        )
+
+    def get_url(self) -> str:
+        """
+        gets the base_url from potential_urls based on configurations
+        """
         potential_urls = URLS[(self.sandbox, self.european)]
         stream_name = "Account"
         for url_prefix in potential_urls:
-            if rest:
-                resp = self._retryable_request("GET","{}v1/describe/{}".format(url_prefix, stream_name), headers=self.rest_headers)
+            if self.is_rest:
+                resp = self._retryable_request(
+                    "GET",
+                    f"{url_prefix}v1/describe/{stream_name}", url_check=True,
+                    headers=self.rest_headers,
+                )
+
             else:
-                query = "select * from {} limit 1".format(stream_name)
-                post_url = "{}v1/batch-query/".format(url_prefix)
-                payload = Aqua.make_payload(stream_name, "discover", query, self.partner_id)
-                resp = self._retryable_request("POST", post_url, auth=self.aqua_auth, json=payload)
-                if resp.status_code==200:
+                query = f"select * from {stream_name} limit 1"
+                post_url = f"{url_prefix}v1/batch-query/"
+                payload = make_payload("discover", query, self.partner_id)
+                resp = self._retryable_request(
+                    "POST", post_url, url_check=True, auth=self.aqua_auth, json=payload
+                )
+                if resp.status_code == 200:
                     resp_json = resp.json()
-                    delete_id = resp_json['id']
-                    delete_url = "{}v1/batch-query/jobs/{}".format(url_prefix, delete_id)
-                    self._retryable_request("DELETE", delete_url, auth=self.aqua_auth)
+                    # Checks whether there is an error in http response
+                    if "errorCode" not in resp_json:
+                        delete_id = resp_json["id"]
+                        delete_url = f"{url_prefix}v1/batch-query/jobs/{delete_id}"
+                        self._retryable_request(
+                            "DELETE", delete_url, auth=self.aqua_auth
+                        )
+                    else:
+                        LOGGER.error(resp_json.get("message"))
```

Comment:
This should actually be an exception but not error logging because Zuora is sending 200 status code for invalid partner Id with message `Partner ID is not recognized. To obtain a partner ID, submit a request with Zuora Global Support.` So replaced the error logger with exception